{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy import stats\n",
    "import itertools\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#################################################\n",
    "#### ClueGO cyREST Python 3 Example Workflow ####\n",
    "#################################################\n",
    "\n",
    "# In Windows install first python for Windows 3.7.2 from https://www.python.org/downloads/release/python-372/\n",
    "#### Please make sure that Cytoscape v3.6+' is started and the Cytoscape Apps 'yFiles Layout Algorithms' and 'ClueGO v2.5.2' are installed before running this script! ####\n",
    "\n",
    "# required libraries that need to be installed before starting the script\n",
    "# In Windows install first pip with e.g. py get-pip.py  (https://bootstrap.pypa.io/get-pip.py)\n",
    "import requests # to install use \"pip install requests\" or \"sudo apt install python3-request\" in ubuntu linux\n",
    "import json\n",
    "import os\n",
    "import csv\n",
    "from pathlib import Path\n",
    "from urllib.parse import quote\n",
    "import time\n",
    "\n",
    "# Helper functions\n",
    "def read_gene_list(data):\n",
    "    gene_ids=[]\n",
    "    with open(data,'r') as f:\n",
    "        reader=csv.reader(f,delimiter='\\t')\n",
    "        for row in reader:\n",
    "            gene_ids.append(row[0])\n",
    "    return gene_ids\n",
    "\n",
    "def writeLines(lines,out_file):\n",
    "    file = open(out_file,'w')\n",
    "    for line in lines:\n",
    "        file.write(line)\n",
    "    file.close()\n",
    "\n",
    "def writeBin(raw,out_file):\n",
    "    file = open(out_file,'wb')\n",
    "    file.write(raw)\n",
    "    file.close()\n",
    "\n",
    "def dframe_stack_list(ins,col,typ='float32'):\n",
    "    df = ins.copy(deep=True)\n",
    "    s = df[col].apply(lambda x: pd.Series(list(x)),1).stack().astype(typ)\n",
    "    s.index = s.index.droplevel(-1)\n",
    "    s.name = col\n",
    "    del df[col]\n",
    "    df = df.join(s)\n",
    "    return df\n",
    "\n",
    "def fix_pvals(x):\n",
    "    if x < 1e-16:\n",
    "        x=1e-16\n",
    "    return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Example analysis Parameters for ClueGO_output\n",
      "User Home Folder: /Users/zrimec\n",
      "User Output Folder: /Users/zrimec/ClueGO_output\n",
      "Cytoscape Base URL: http://localhost:1234/v1\n",
      "ClueGO Base URL: http://localhost:1234/v1/apps/cluego/cluego-manager\n"
     ]
    }
   ],
   "source": [
    "#### Basic settings for cyREST ####\n",
    "# home.folder = \"/home/user\"\n",
    "EXAMPLE_SELECTION = \"ClueGO_output\"\n",
    "HOME_FOLDER = str(Path.home())\n",
    "\n",
    "# # Windows specific command to set home folder to user-home\n",
    "# HOME_FOLDER = HOME_FOLDER.replace(\"\\\\Documents\",\"\")\n",
    "\n",
    "SEP = \"/\"\n",
    "OUTPUT_FOLDER = HOME_FOLDER+SEP+EXAMPLE_SELECTION\n",
    "try:\n",
    "    os.stat(OUTPUT_FOLDER)\n",
    "except:\n",
    "    os.mkdir(OUTPUT_FOLDER)\n",
    "\n",
    "CLUEGO_HOME_FOLDER = HOME_FOLDER+SEP+\"ClueGOConfiguration\"+SEP+\"v2.5.7\"\n",
    "PORT_NUMBER = \"1234\"\n",
    "HOST_ADDRESS = \"localhost\"\n",
    "HEADERS = {'Content-Type': 'application/json'}\n",
    "\n",
    "# define base urls\n",
    "CYTOSCAPE_BASE_URL = \"http://\"+HOST_ADDRESS+\":\"+PORT_NUMBER+SEP+\"v1\"\n",
    "CLUEGO_BASE_URL = CYTOSCAPE_BASE_URL+SEP+\"apps\"+SEP+\"cluego\"+SEP+\"cluego-manager\"\n",
    "\n",
    "print(\"Example analysis Parameters for \"+EXAMPLE_SELECTION)\n",
    "print(\"User Home Folder: \"+HOME_FOLDER)\n",
    "print(\"User Output Folder: \"+OUTPUT_FOLDER)\n",
    "print(\"Cytoscape Base URL: \"+CYTOSCAPE_BASE_URL)\n",
    "print(\"ClueGO Base URL: \"+CLUEGO_BASE_URL)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7020, 13)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>conditions</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sheet</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ac_for</th>\n",
       "      <td>18</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fur_HMF</th>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hydrolysates</th>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenolics</th>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              strain  conditions\n",
       "sheet                           \n",
       "ac_for            18          39\n",
       "fur_HMF           12          18\n",
       "hydrolysates      17          19\n",
       "phenolics          6          16"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in all the data\n",
    "# append from different sheets to one df\n",
    "\n",
    "sheet_names = ['acetate',\n",
    "          'furfural',\n",
    "          'HMF',\n",
    "          'formic acid',\n",
    "          'phenolics',\n",
    "          'hydrolysates']\n",
    "\n",
    "df = []\n",
    "for i in range(6):\n",
    "    df.append(pd.read_excel('../data/210310_List target genes_EC.xlsx',sheet_name=i))\n",
    "    df[-1]['sheet'] = sheet_names[i]\n",
    "    if i==0:\n",
    "        df[-1] = df[-1].drop(0)\n",
    "df = pd.concat(df)\n",
    "# fix col names\n",
    "df.columns = pd.Series(df.columns).str.lower().str.replace(' ','_')\n",
    "print(df.shape)\n",
    "df.head()\n",
    "\n",
    "# combine groups\n",
    "df['sheet'] = df.sheet.map({'acetate':'ac_for',\n",
    "                  'formic acid':'ac_for',\n",
    "                  'HMF':'fur_HMF',\n",
    "                  'furfural':'fur_HMF',\n",
    "                  'hydrolysates':'hydrolysates',\n",
    "                  'phenolics':'phenolics'\n",
    "                 })\n",
    "df.groupby('sheet').gene.count()\n",
    "df.groupby('sheet')['strain','conditions'].nunique()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6604, 16)\n",
      "0.0065527065527065526\n",
      "0.1284900284900285\n",
      "0.8021367521367522\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Primary_SGDID</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature_qualifier</th>\n",
       "      <th>Feature_name</th>\n",
       "      <th>Standard_gene_name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Parent_feature_name</th>\n",
       "      <th>Secondary_SGDID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_coordinate</th>\n",
       "      <th>Stop_coordinate</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Genetic_position</th>\n",
       "      <th>Coordinate_version</th>\n",
       "      <th>Sequence_version</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S000002143</td>\n",
       "      <td>ORF</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chromosome 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>335.0</td>\n",
       "      <td>649.0</td>\n",
       "      <td>W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1996-07-31</td>\n",
       "      <td>1996-07-31</td>\n",
       "      <td>Dubious open reading frame; unlikely to encode...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S000028594</td>\n",
       "      <td>ORF</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>YAL068W-A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chromosome 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>538.0</td>\n",
       "      <td>792.0</td>\n",
       "      <td>W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2003-07-29</td>\n",
       "      <td>2003-07-29</td>\n",
       "      <td>Dubious open reading frame; unlikely to encode...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S000002142</td>\n",
       "      <td>ORF</td>\n",
       "      <td>Verified</td>\n",
       "      <td>YAL068C</td>\n",
       "      <td>PAU8</td>\n",
       "      <td>seripauperin PAU8</td>\n",
       "      <td>chromosome 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2169.0</td>\n",
       "      <td>1807.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1996-07-31</td>\n",
       "      <td>1996-07-31</td>\n",
       "      <td>Protein of unknown function; member of the ser...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S000028593</td>\n",
       "      <td>ORF</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>YAL067W-A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chromosome 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>2480.0</td>\n",
       "      <td>2707.0</td>\n",
       "      <td>W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2003-07-29</td>\n",
       "      <td>2003-07-29</td>\n",
       "      <td>Putative protein of unknown function; identifi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S000000062</td>\n",
       "      <td>ORF</td>\n",
       "      <td>Verified</td>\n",
       "      <td>YAL067C</td>\n",
       "      <td>SEO1</td>\n",
       "      <td>putative permease SEO1</td>\n",
       "      <td>chromosome 1</td>\n",
       "      <td>L000003363</td>\n",
       "      <td>1</td>\n",
       "      <td>9016.0</td>\n",
       "      <td>7235.0</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2011-02-03</td>\n",
       "      <td>1996-07-31</td>\n",
       "      <td>Putative permease; member of the allantoate tr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Primary_SGDID Feature_type Feature_qualifier Feature_name  \\\n",
       "0     S000002143          ORF           Dubious      YAL069W   \n",
       "2     S000028594          ORF           Dubious    YAL068W-A   \n",
       "9     S000002142          ORF          Verified      YAL068C   \n",
       "11    S000028593          ORF   Uncharacterized    YAL067W-A   \n",
       "14    S000000062          ORF          Verified      YAL067C   \n",
       "\n",
       "   Standard_gene_name                   Alias Parent_feature_name  \\\n",
       "0                 NaN                     NaN        chromosome 1   \n",
       "2                 NaN                     NaN        chromosome 1   \n",
       "9                PAU8       seripauperin PAU8        chromosome 1   \n",
       "11                NaN                     NaN        chromosome 1   \n",
       "14               SEO1  putative permease SEO1        chromosome 1   \n",
       "\n",
       "   Secondary_SGDID Chromosome  Start_coordinate  Stop_coordinate Strand  \\\n",
       "0              NaN          1             335.0            649.0      W   \n",
       "2              NaN          1             538.0            792.0      W   \n",
       "9              NaN          1            2169.0           1807.0      C   \n",
       "11             NaN          1            2480.0           2707.0      W   \n",
       "14      L000003363          1            9016.0           7235.0      C   \n",
       "\n",
       "    Genetic_position Coordinate_version Sequence_version  \\\n",
       "0                NaN         1996-07-31       1996-07-31   \n",
       "2                NaN         2003-07-29       2003-07-29   \n",
       "9                NaN         1996-07-31       1996-07-31   \n",
       "11               NaN         2003-07-29       2003-07-29   \n",
       "14               NaN         2011-02-03       1996-07-31   \n",
       "\n",
       "                                          Description  \n",
       "0   Dubious open reading frame; unlikely to encode...  \n",
       "2   Dubious open reading frame; unlikely to encode...  \n",
       "9   Protein of unknown function; member of the ser...  \n",
       "11  Putative protein of unknown function; identifi...  \n",
       "14  Putative permease; member of the allantoate tr...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load yeast gene names and GO slims\n",
    "# folder = '../../DeepExpression/paper-z1/scripts/'\n",
    "# fn1 = 'SGD_features.tab'\n",
    "\n",
    "import wget\n",
    "url = 'https://downloads.yeastgenome.org/curation/chromosomal_feature/SGD_features.tab'\n",
    "fname = wget.download(url)\n",
    "columns = ['Primary_SGDID',\n",
    "           'Feature_type',\n",
    "            'Feature_qualifier',\n",
    "            'Feature_name',\n",
    "            'Standard_gene_name',\n",
    "            'Alias', # (optional, multiples separated by |)\n",
    "            'Parent_feature_name',\n",
    "            'Secondary_SGDID',\n",
    "            'Chromosome',\n",
    "            'Start_coordinate',\n",
    "            'Stop_coordinate',\n",
    "            'Strand',\n",
    "            'Genetic_position',\n",
    "            'Coordinate_version',\n",
    "            'Sequence_version',\n",
    "            'Description']\n",
    "gene_function = (pd\n",
    "                 .read_csv(fname,sep='\\t',header=None,names=columns)\n",
    "                 #.set_index('Feature name')\n",
    "                 .query('Feature_type==\"ORF\"')\n",
    "                )\n",
    "print(gene_function.shape)\n",
    "\n",
    "print(sum(df.gene.isin(gene_function.Alias.unique()))/df.shape[0])\n",
    "print(sum(df.gene.isin(gene_function.Feature_name.unique()))/df.shape[0])\n",
    "print(sum(df.gene.isin(gene_function.Standard_gene_name.unique()))/df.shape[0])\n",
    "\n",
    "gene_function.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.935042735042735\n",
      "0.9884615384615385\n"
     ]
    }
   ],
   "source": [
    "# check gene coverage\n",
    "# are there gene names that are not standard?\n",
    "df[~df.gene.isin(gene_function.Standard_gene_name.unique())].gene\n",
    "\n",
    "# combine all name types\n",
    "all_names=list(gene_function.Alias.unique())\n",
    "all_names.extend(list(gene_function.Feature_name.unique()))\n",
    "all_names.extend(list(gene_function.Standard_gene_name.unique()))\n",
    "len(all_names)\n",
    "print(sum(df.gene.isin(all_names))/df.shape[0])\n",
    "df[~df.gene.isin(all_names)].gene\n",
    "\n",
    "# fix gene names to capital final letters as it is in SGD\n",
    "# fix italic gene names => remove \\xa0 string\n",
    "# remove spaces\n",
    "df['gene_fix'] = (df.gene\n",
    "                  .astype(str)\n",
    "                  .apply(lambda x: x.replace('\\xa0','').replace(' ',''))\n",
    "                  .apply(lambda x: x.upper())\n",
    "                 )\n",
    "df.gene_fix\n",
    "\n",
    "# then recheck how many\n",
    "print(sum(df.gene_fix.isin(all_names))/df.shape[0])\n",
    "df[~df.gene_fix.isin(all_names)].gene_fix.tolist()\n",
    "\n",
    "df['gene'] = df.gene_fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inh1</th>\n",
       "      <th>inh2</th>\n",
       "      <th>effect</th>\n",
       "      <th>names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>+</td>\n",
       "      <td>[ACE2, CKB2, ECM30, GIS4, GRE3, GTR2, HAL5, HE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>-</td>\n",
       "      <td>[AIM10, AIM4, ARF1, BCH1, BST1, BUD28, CAT8, C...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>phenolics</td>\n",
       "      <td>+</td>\n",
       "      <td>[ADE1, ADE13, ADE17, ADR1, ALD5, ATF1, ATF2, B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>phenolics</td>\n",
       "      <td>-</td>\n",
       "      <td>[AFG3, ARP6, BUD28, CAT8, CHC1, CHO1, CIK1, EC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>+</td>\n",
       "      <td>[ACF4, ADE1, ADE13, ADE17, ADE6, APA2, ARG2, A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ac_for</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>-</td>\n",
       "      <td>[AAC3, AIM4, AKR1, ALG9, ANP1, APQ12, AQR1, AR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>phenolics</td>\n",
       "      <td>+</td>\n",
       "      <td>[MSN2, YAP1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>phenolics</td>\n",
       "      <td>-</td>\n",
       "      <td>[ZWF1, ADO1, ERG3, ERG4, ERG5, BEM2, SNG1, RPL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>+</td>\n",
       "      <td>[HIS7, HAL5, SAT4, ADH6, CIN5, DIC1, GLR1, GSH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>fur_HMF</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>-</td>\n",
       "      <td>[TPO3, COM2, HRK1, VID24, ZWF1, GOR1, RTG2, TR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>phenolics</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>+</td>\n",
       "      <td>[CRZ1, DAL81, LEU3, MSN2, PAD1, YAP1, ADE1, AD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>phenolics</td>\n",
       "      <td>hydrolysates</td>\n",
       "      <td>-</td>\n",
       "      <td>[ATR1, CAT8, GAL4, GPH1, GZF3, PDR1, PDR8, PUT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         inh1          inh2 effect  \\\n",
       "0      ac_for       fur_HMF      +   \n",
       "1      ac_for       fur_HMF      -   \n",
       "2      ac_for     phenolics      +   \n",
       "3      ac_for     phenolics      -   \n",
       "4      ac_for  hydrolysates      +   \n",
       "5      ac_for  hydrolysates      -   \n",
       "6     fur_HMF     phenolics      +   \n",
       "7     fur_HMF     phenolics      -   \n",
       "8     fur_HMF  hydrolysates      +   \n",
       "9     fur_HMF  hydrolysates      -   \n",
       "10  phenolics  hydrolysates      +   \n",
       "11  phenolics  hydrolysates      -   \n",
       "\n",
       "                                                names  \n",
       "0   [ACE2, CKB2, ECM30, GIS4, GRE3, GTR2, HAL5, HE...  \n",
       "1   [AIM10, AIM4, ARF1, BCH1, BST1, BUD28, CAT8, C...  \n",
       "2   [ADE1, ADE13, ADE17, ADR1, ALD5, ATF1, ATF2, B...  \n",
       "3   [AFG3, ARP6, BUD28, CAT8, CHC1, CHO1, CIK1, EC...  \n",
       "4   [ACF4, ADE1, ADE13, ADE17, ADE6, APA2, ARG2, A...  \n",
       "5   [AAC3, AIM4, AKR1, ALG9, ANP1, APQ12, AQR1, AR...  \n",
       "6                                        [MSN2, YAP1]  \n",
       "7   [ZWF1, ADO1, ERG3, ERG4, ERG5, BEM2, SNG1, RPL...  \n",
       "8   [HIS7, HAL5, SAT4, ADH6, CIN5, DIC1, GLR1, GSH...  \n",
       "9   [TPO3, COM2, HRK1, VID24, ZWF1, GOR1, RTG2, TR...  \n",
       "10  [CRZ1, DAL81, LEU3, MSN2, PAD1, YAP1, ADE1, AD...  \n",
       "11  [ATR1, CAT8, GAL4, GPH1, GZF3, PDR1, PDR8, PUT...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# prepare overlapping gene lists\n",
    "\n",
    "gene_lists = []\n",
    "for inh1,inh2 in list(itertools.combinations(df.sheet.unique(),2)):\n",
    "    for eff in ['+','-']:\n",
    "        if inh1!=inh2:\n",
    "            # prepare test set\n",
    "            sel1 = (df\n",
    "                 .query('sheet==@inh1')\n",
    "                 .query('growth_effect==@eff'))\n",
    "            sel2 = (df\n",
    "                 .query('sheet==@inh2')\n",
    "                 .query('growth_effect==@eff'))\n",
    "            gene_lists.append([inh1,inh2,eff,pd.DataFrame(sel1[sel1.gene.isin(sel2.gene)]\n",
    "                        .gene.drop_duplicates())\n",
    "                        #.merge(df_go,on='gene',how='left')\n",
    "                        #.dropna(subset=['gene','entrez_id'])\n",
    "                        #.entrez_id\n",
    "                        .gene\n",
    "                        .tolist()])\n",
    "gene_lists = pd.DataFrame(gene_lists,columns=['inh1','inh2','effect','names'])\n",
    "gene_lists"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run ClueGO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_cluego(OUTPUT_FOLDER):\n",
    "    #### 0.0 Start up ClueGO in case it is not running yet\n",
    "    response = requests.post(CYTOSCAPE_BASE_URL+SEP+\"apps\"+SEP+\"cluego\"+SEP+\"start-up-cluego\", headers=HEADERS)\n",
    "    # wait 2 seconds to make sure ClueGO is started\n",
    "    if(str(response)==\"<Response [500]>\"):\n",
    "        print(\"wait 2 secs\")\n",
    "        time.sleep(2) \n",
    "\n",
    "    #### 1.0 Select the ClueGO Organism to analyze ####\n",
    "    organism_name = \"Saccharomyces cerevisiae S288c\" # (run \"1.1 Get all ClueGO organisms\" to get all options)\n",
    "    print(\"1.0 Select the ClueGO Organism to analyze: \"+organism_name)\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"organisms\"+SEP+\"set-organism\"+SEP+quote(organism_name), headers=HEADERS)\n",
    "\n",
    "    ## [optional functions and settings, un comment and modify if needed]\n",
    "    #  # 1.1 Get all ClueGO organisms\n",
    "    #response <- requests.get(CLUEGO_BASE_URL+SEP+\"organisms\"+SEP+\"get-all-installed-organisms\")\n",
    "    #print(response.json())\n",
    "\n",
    "    # # 1.2 Get all info for installed organisms\n",
    "    #response <- requests.get(CLUEGO_BASE_URL+SEP+\"organisms\"+SEP+\"get-all-organism-info\")\n",
    "    #print(response.json())\n",
    "\n",
    "    #### 2.0 Upload IDs for a specific Cluster ####\n",
    "    print(\"2.0 Upload IDs for a specific Cluster\")\n",
    "\n",
    "    cluster1 = 1\n",
    "    cluster2 = 2\n",
    "\n",
    "    # upload gene list for cluster 1\n",
    "    FILE_LOCATION = OUTPUT_FOLDER+\"/gene_list.txt\"\n",
    "    gene_list1 = json.dumps(read_gene_list(FILE_LOCATION))\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"upload-ids-list\"+SEP+quote(str(cluster1)), data=gene_list1, headers=HEADERS)\n",
    "\n",
    "    # 2.1 Set the number of Clusters\n",
    "    max_input_panel_number = 1\n",
    "    if(EXAMPLE_SELECTION == \"ClueGO Rest Example for two gene lists\"):\n",
    "        max_input_panel_number = 2\n",
    "\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"max-input-panel\"+SEP+str(max_input_panel_number))\n",
    "\n",
    "    if(EXAMPLE_SELECTION == \"ClueGO Rest Example for two gene lists\"):\n",
    "        # upload gene list for cluster 2\n",
    "\n",
    "        FILE_LOCATION = CLUEGO_HOME_FOLDER+SEP+\"ClueGOExampleFiles/GSE6887_NKcell_Healthy_top200UpRegulated.txt\"\n",
    "        gene_list2 = json.dumps(read_gene_list(FILE_LOCATION))\n",
    "        response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"upload-ids-list\"+SEP+quote(str(cluster2)), data=gene_list2, headers=HEADERS)\n",
    "\n",
    "    # To add here in the same way if you have more than 2 cluster to compare\n",
    "\n",
    "    ## [optional functions and settings, un comment and modify if needed]\n",
    "    ## 2.2 Select the ClueGO ID type\n",
    "    #id_type_name = \"# Automatic #\" # (run \"2.3 Get ClueGO ID types\" to get all options)\n",
    "    #response = requests.put(CLUEGO_BASE_URL+SEP+\"ids\"+SEP+\"set-id-type\"+SEP+id_type_name, headers=HEADERS)\n",
    "\n",
    "    ## 2.3 Refresh ClueGO source files\n",
    "    #requests.post(CLUEGO_BASE_URL+SEP+\"ids\"+SEP+\"refresh-cluego-id-files\", headers=HEADERS)\n",
    "\n",
    "    ## 2.4 Get ClueGO ID types\n",
    "    #response = requests.get(CLUEGO_BASE_URL+SEP+\"ids\"+SEP+\"get-all-installed-id-types\", headers=HEADERS)\n",
    "    #print(response.json())\n",
    "\n",
    "    # 2.5 Set analysis properties for a Cluster, this is to repeat for each input cluster\n",
    "    input_panel_index = cluster1 # set here the cluster number e.g. \"1\"\n",
    "    node_shape = \"Ellipse\" # (\"Ellipse\",\"Diamond\",\"Hexagon\",\"Octagon\",\"Parallelogram\",\"Rectangle\",\"Round Rectangle\",\"Triangle\",\"V\")\n",
    "    cluster_color = \"#ff0000\" # The color in hex, e.g. #F3A455\n",
    "    min_number_of_genes_per_term = 3\n",
    "    min_percentage_of_genes_mapped = 4\n",
    "    no_restrictions = False # \"True\" for no restricions in number and percentage per term\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"set-analysis-properties\"+SEP+str(input_panel_index)+SEP+node_shape+SEP+quote(cluster_color)+SEP+str(min_number_of_genes_per_term)+SEP+str(min_percentage_of_genes_mapped)+SEP+str(no_restrictions), headers=HEADERS)\n",
    "\n",
    "    if(EXAMPLE_SELECTION == \"ClueGO Rest Example for two gene lists\"):\n",
    "        input_panel_index = cluster2 # set here the cluster number e.g. \"2\"\n",
    "        node_shape = \"Ellipse\" # (\"Ellipse\",\"Diamond\",\"Hexagon\",\"Octagon\",\"Parallelogram\",\"Rectangle\",\"Round Rectangle\",\"Triangle\",\"V\")\n",
    "        cluster_color = \"#0000ff\" # The color in hex, e.g. #F3A455\n",
    "        min_number_of_genes_per_term = 3\n",
    "        min_percentage_of_genes_mapped = 4\n",
    "        no_restrictions = False # \"True\" for no restricions in number and percentage per term\n",
    "        response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"set-analysis-properties\"+SEP+str(input_panel_index)+SEP+node_shape+SEP+quote(cluster_color)+SEP+str(min_number_of_genes_per_term)+SEP+str(min_percentage_of_genes_mapped)+SEP+str(no_restrictions), headers=HEADERS)\n",
    "\n",
    "    # 2.6 Select visual style\n",
    "    if(EXAMPLE_SELECTION == \"ClueGO Rest Example for two gene lists\"):\n",
    "        visual_style = \"ShowClusterDifference\"  # (ShowGroupDifference, ShowSignificanceDifference, ShowClusterDifference)\n",
    "    else:\n",
    "        visual_style = \"ShowGroupDifference\"  # (ShowGroupDifference, ShowSignificanceDifference, ShowClusterDifference)\n",
    "\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"cluster\"+SEP+\"select-visual-style\"+SEP+visual_style, headers=HEADERS)\n",
    "\n",
    "    ####  3.0 Select Ontologies\n",
    "    print(\"3.0 Select Ontologies\")\n",
    "\n",
    "    selected_ontologies = json.dumps([\"3;Ellipse\",\"5;Ellipse\",\"7;Ellipse\"]) # (run \"3.1 Get all available Ontologies\" to get all options)\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+\"set-ontologies\", data=selected_ontologies, headers=HEADERS)\n",
    "\n",
    "    ## [optional functions and settings, un comment and modify if needed]\n",
    "    ## 3.1 Get all available Ontologies\n",
    "    #response = requests.get(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+\"get-ontology-info\", headers=HEADERS)\n",
    "    #print(response.json())\n",
    "\n",
    "    ## 3.2 Select Evidence Codes\n",
    "    #evidence_codes = json.dumps([\"All\"]) # (run \"3.3 Get all available Evidence Codes\" to get all options)\n",
    "    #response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+\"set-evidence-codes\", data=evidence_codes, headers=HEADERS)\n",
    "\n",
    "    ## 3.3 Get all available Evidence Codes\n",
    "    #response = requests.get(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+\"get-evidence-code-info\", headers=HEADERS)\n",
    "    #print(response.json())\n",
    "\n",
    "    # 3.4 Set min, max GO tree level\n",
    "    min_level = 3\n",
    "    max_level = 8\n",
    "    all_levels = False\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+\"set-min-max-levels\"+SEP+str(min_level)+SEP+str(max_level)+SEP+str(all_levels), headers=HEADERS)\n",
    "\n",
    "    # 3.5 Use GO term significance cutoff\n",
    "    p_value_cutoff = 0.05\n",
    "    use_significance_cutoff = True\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+str(use_significance_cutoff)+SEP+str(p_value_cutoff), headers=HEADERS)\n",
    "\n",
    "    # 3.6 Use GO term fusion\n",
    "    use_go_term_fusion = True\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\"+SEP+str(use_go_term_fusion), headers=HEADERS)\n",
    "\n",
    "    # 3.7 Set statistical parameters\n",
    "    enrichment_type = \"Enrichment/Depletion (Two-sided hypergeometric test)\" # (\"Enrichment (Right-sided hypergeometric test)\", \"Depletion (Left-sided hypergeometric test)\", \"Enrichment/Depletion (Two-sided hypergeometric test)\")\n",
    "    multiple_testing_correction = True #Note: manually set BH-correction\n",
    "    use_mid_pvalues = False\n",
    "    use_doubling = False\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"stats\"+SEP+enrichment_type+SEP+str(multiple_testing_correction)+SEP+str(use_mid_pvalues)+SEP+str(use_doubling), headers=HEADERS)\n",
    "\n",
    "    # 3.8 Set the Kappa Score level\n",
    "    kappa_score = 0.4\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"ontologies\",\"set-kappa-score-level\"+SEP+str(kappa_score), headers=HEADERS)\n",
    "\n",
    "    # 3.9 Set grouping parameters\n",
    "    do_grouping = True\n",
    "    coloring_type = \"Random\" # (\"Random\",\"Fix\")\n",
    "    group_leading_term = \"Highest Significance\" # (\"Highest Significance\",\"#Genes / Term\",\"%Genes / Term\",\"%Genes / Term vs Cluster\")\n",
    "    grouping_type = \"Kappa Score\" # (\"Kappa Score\",\"Tree\")\n",
    "    init_group_size = 1\n",
    "    perc_groups_for_merge = 50\n",
    "    perc_terms_for_merge = 50\n",
    "    response = requests.put(CLUEGO_BASE_URL+SEP+\"grouping\"+SEP+str(do_grouping)+SEP+coloring_type+SEP+group_leading_term+SEP+grouping_type+SEP+str(init_group_size)+SEP+str(perc_groups_for_merge)+SEP+str(perc_terms_for_merge), headers=HEADERS)\n",
    "\n",
    "    #### 4.0 Run ClueGO Analysis ####\n",
    "    print(\"4.0 Run ClueGO Analysis\")\n",
    "\n",
    "    # Run the analysis an save log file\n",
    "    analysis_name = EXAMPLE_SELECTION\n",
    "    analysis_option = \"Cancel and refine selection\" # (\"Continue analysis\",\"Skip the grouping\",\"Cancel and refine selection\")  -> Analysis option in case there are more than 1000 terms found!\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+quote(analysis_name)+SEP+quote(analysis_option), headers=HEADERS)\n",
    "    log_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Example-Analysis-log.txt\"\n",
    "    writeLines(response.text,log_file_name)\n",
    "    #print(response.text)\n",
    "\n",
    "    # 4.1 Get network id (SUID) (CyRest function from Cytoscape)\n",
    "    response = requests.get(CYTOSCAPE_BASE_URL+SEP+\"networks\"+SEP+\"currentNetwork\", headers=HEADERS)\n",
    "    current_network_suid = response.json()['data']['networkSUID']\n",
    "    #print(current_network_suid)\n",
    "\n",
    "    print(\"Save results\")\n",
    "\n",
    "    # Get network graphics (CyRest function from Cytoscape)\n",
    "    image_type = \"svg\" # png, pdf\n",
    "    response = requests.get(CYTOSCAPE_BASE_URL+SEP+\"networks\"+SEP+str(current_network_suid)+SEP+\"views\"+SEP+\"first.\"+image_type)\n",
    "    image_file_name = OUTPUT_FOLDER+SEP+\"ClueGOExampleNetwork.\"+image_type\n",
    "    writeBin(response.content,image_file_name)\n",
    "\n",
    "    # 4.2 Get ClueGO result table\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-cluego-table\"+SEP+str(current_network_suid))\n",
    "    table_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Example-Result-Table.txt\"\n",
    "    writeLines(response.text,table_file_name)\n",
    "\n",
    "    # 4.3 Get ClueGO genes and main functions\n",
    "    number_of_functions_to_add = 3\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-main-functions\"+SEP+str(current_network_suid)+SEP+str(number_of_functions_to_add))\n",
    "    table_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Genes-With-Main-Functions.txt\"\n",
    "    writeLines(response.text,table_file_name)\n",
    "\n",
    "    # 4.4 Get genes result table\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-gene-table\"+SEP+str(current_network_suid))\n",
    "    table_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Gene-Table.txt\"\n",
    "    writeLines(response.text,table_file_name)\n",
    "\n",
    "    # 4.5 Get Kappascore Matrix\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-kappascore-matrix\"+SEP+str(current_network_suid))\n",
    "    table_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Kappascore-Matrix.txt\"\n",
    "    writeLines(response.text,table_file_name)\n",
    "\n",
    "    # 4.6 Get binary Gene-Term Matrix\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-binary-gene-term-matrix\"+SEP+str(current_network_suid))\n",
    "    table_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-Binary-Gene-Term-Matrix.txt\"\n",
    "    writeLines(response.text,table_file_name)\n",
    "\n",
    "    # 4.7 ClueGO Result Chart\n",
    "    # Get result charts for both cluster as pie chart\n",
    "    chart_type = \"PieChart\" # (\"PieChart\",\"BarChart\")\n",
    "    image_type = \"svg\" # (\"svg\",\"png\",\"pdf\")\n",
    "    response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-cluego-result-chart\"+SEP+str(current_network_suid)+SEP+str(cluster1)+SEP+chart_type+SEP+image_type)\n",
    "    image_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-\"+chart_type+\"-For-Cluster\"+str(cluster1)+\".\"+image_type\n",
    "    writeBin(response.content,image_file_name)\n",
    "\n",
    "    if(EXAMPLE_SELECTION == \"ClueGO Rest Example for two gene lists\"):\n",
    "        response = requests.get(CLUEGO_BASE_URL+SEP+\"analysis-results\"+SEP+\"get-cluego-result-chart\"+SEP+str(current_network_suid)+SEP+str(cluster2)+SEP+chart_type+SEP+image_type)\n",
    "        image_file_name = OUTPUT_FOLDER+SEP+\"ClueGO-\"+chart_type+\"-For-Cluster\"+str(cluster2)+\".\"+image_type\n",
    "        writeBin(response.content,image_file_name)\n",
    "\n",
    "    ### [optional functions and settings, un comment and modify if needed]\n",
    "    ## 4.8 Remove ClueGO analysis result\n",
    "    #print(\"Remove ClueGO Network\")\n",
    "    ## Remove analysis to reduce memory usage. This is important when using patch modes that create lots of analyses.\n",
    "    #response = requests.delete(CLUEGO_BASE_URL+SEP+\"remove-cluego-analysis-result\"+SEP+str(current_network_suid))\n",
    "\n",
    "    ## 4.9 Remove all ClueGO analysis results\n",
    "    #print(\"Remove all ClueGO Networks\")\n",
    "    #response = requests.delete(CLUEGO_BASE_URL+SEP+\"remove-all-cluego-analysis-results\")\n",
    "\n",
    "    print(\"Example analysis \"+EXAMPLE_SELECTION+\" done\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_fur_HMF_+: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_fur_HMF_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_fur_HMF_-: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_fur_HMF_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_phenolics_+: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_phenolics_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_phenolics_-: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_phenolics_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_hydrolysates_+: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_hydrolysates_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/ac_for_hydrolysates_-: File exists\n",
      "/Users/zrimec/ClueGO_output/ac_for_hydrolysates_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/fur_HMF_phenolics_+: File exists\n",
      "/Users/zrimec/ClueGO_output/fur_HMF_phenolics_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/fur_HMF_phenolics_-: File exists\n",
      "/Users/zrimec/ClueGO_output/fur_HMF_phenolics_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/fur_HMF_hydrolysates_+: File exists\n",
      "/Users/zrimec/ClueGO_output/fur_HMF_hydrolysates_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/fur_HMF_hydrolysates_-: File exists\n",
      "/Users/zrimec/ClueGO_output/fur_HMF_hydrolysates_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/phenolics_hydrolysates_+: File exists\n",
      "/Users/zrimec/ClueGO_output/phenolics_hydrolysates_+\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n",
      "mkdir: /Users/zrimec/ClueGO_output/phenolics_hydrolysates_-: File exists\n",
      "/Users/zrimec/ClueGO_output/phenolics_hydrolysates_-\n",
      "1.0 Select the ClueGO Organism to analyze: Saccharomyces cerevisiae S288c\n",
      "2.0 Upload IDs for a specific Cluster\n",
      "3.0 Select Ontologies\n",
      "4.0 Run ClueGO Analysis\n",
      "Save results\n",
      "Example analysis ClueGO_output done\n"
     ]
    }
   ],
   "source": [
    "for idx,row in gene_lists.iterrows():\n",
    "    OUTPUT_FOLDER2 = OUTPUT_FOLDER+'/'+row.inh1+'_'+row.inh2+'_'+row.effect\n",
    "    !mkdir {OUTPUT_FOLDER2}\n",
    "    print(OUTPUT_FOLDER2)\n",
    "    \n",
    "    # save to list of genes newline\n",
    "    with open(OUTPUT_FOLDER2+\"/gene_list.txt\", \"w\") as f:\n",
    "        for i in row.names:\n",
    "            f.write(i+'\\n')\n",
    "            \n",
    "    run_cluego(OUTPUT_FOLDER2)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "hydrolysates",
   "language": "python",
   "name": "hydrolysates"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
